MultiServer.py

```diff
@@ -1504,6 +1519,9 @@ def get_hints(self, input_text: str, for_location: bool = False) -> bool:
                 self.ctx.random.shuffle(not_found_hints)
                 # By popular vote, make hints prefer non-local placements
                 not_found_hints.sort(key=lambda hint: int(hint.receiving_player != hint.finding_player))
+                # By another popular vote, prefer early sphere
+                not_found_hints.sort(key=lambda hint: self.ctx.get_sphere(hint.finding_player, hint.location),
+                                     reverse=True)
```

Comment:
Was there any discussion about prioritizing early spheres over non-local placements (as it looks like this is doing)?

If there's an earlier sphere in my world, and a later sphere in someone else's world,
which do players prefer?

Reply:
Depends on the game, I think.  I'll frequently hint QOL stuff in Factorio because it makes my build easier. If I have research for progressive (long handed) inserters accessible within my world, I definitely want to know it.

Reply:
yeah, I hint QOL stuff a lot too. Though IDK if there's generally many "QOL" things I hint that have *multiple of the same hint* in any game; maybe tools in stardew, but that's not always QOL, sometimes it's necessary.

I think the approach of "trying to get unstuck" as a default behavior makes good sense.

Reply:
Sometimes I worry that when I hint something in another players world, that it might be an annoyance or a chore that I'm putting them through.
But for me personally, I like it when another player hints something in my world. It makes it feel like there's more interaction (when sometimes the interaction feels lacking).
While if I hint something, and it's in my own world, there's no interaction with other players that comes from that.

So this could be another consideration on this question.